DataFormats/RecoCandidate/src/IsoDeposit.cc

```diff
@@ -18,12 +18,14 @@ IsoDeposit::IsoDeposit(double eta, double phi) : theDirection(Direction(eta, phi
 
 void IsoDeposit::addDeposit(double dr, double value) {
   Distance relDir = {float(dr), 0.f};
-  theDeposits.insert(std::make_pair(relDir, value));
+  theDeposits.insert(std::lower_bound(theDeposits.begin(), theDeposits.end(), relDir, Compare()),
+                     std::make_pair(relDir, value));
```

Comment:
Without knowing how this is used upstream, I had thought to `push_back` here and then add a `sorted_` flag to the class, where it would only sort on first iteration. But that could well be premature optimization.